### ISSUE TITLE
Not reflecting a function doesn't yield a proper error message

### ISSUE THREAD

**Issue by CodePeters** [ucsd-progsys/liquidhaskell#1338]
I copy the following code from slack:

```haskell
{-@ LIQUID "--reflection"                     @-}
{-@ LIQUID "--ple"                                @-}

import Language.Haskell.Liquid.ProofCombinators 

{-reflect foo -}
{-@ foo  :: Nat ->  xs: [Nat] -> Nat -> Bool / [len xs ]@-}
foo :: Int ->  [Int] -> Int -> Bool
foo lo [] hi     = (lo<=hi)
foo lo (x:xs) hi = (lo<=hi) && foo lo xs x

{-@ empty_foo :: lo:Nat -> hi:{Nat | lo <= hi } -> {foo lo [] hi} @-}
empty_foo :: Int -> Int -> Proof
empty_foo lo hi = (foo lo [] hi) ==. (lo <= hi) ==.True  *** QED
```
The problem with the code above was that the simple theorem empty_foo couldn't be proved because foo function was not properly reflected (symbol @ was missing before and after word foo inside the specification) .So shouldn't a proper error message be produced in order to give some warning that the function is not reflected ??

**Comment by ranjitjhala** [comment]
This is somehow tickled by the edits to `ghcSpecEnv` done in 

   https://github.com/ucsd-progsys/liquidhaskell/pull/1223

which, for some reason that I am currently investigating, just dumps _all_ Haskell definitions into the refinement environment...

**Comment by ranjitjhala** [comment]
Fixed by PR #1341
